src/backend/utils/mmgr/memprot.c

```diff
@@ -208,6 +211,23 @@ void GPMemoryProtect_Shutdown()
 	VmemTracker_Shutdown();
 }
 
+/*
+ * GPMemoryProtect_RequestAddinStartupMemory
+ *		Request that GPMemoryProtect_TrackStartupMemory will register additional
+ *		memory for use by a loadable module.
+ *
+ * This is only useful if called from the _PG_init hook of a library that
+ * is loaded into the postmaster via shared_preload_libraries.
+ */
+void
+GPMemoryProtect_RequestAddinStartupMemory(Size size)
+{
+	if (IsUnderPostmaster || !startup_mem_addin_request_allowed)
+		return;					/* too late */
```

Comment:
how about error?
```suggestion
	if (IsUnderPostmaster) return;
	if (!startup_mem_addin_request_allowed) elog(ERROR, "too late");
```

Reply:
Similar functions, like `RequestAddinShmemSpace()`, do not drop an error in such case:
```
 * ... Once
 * shared memory has been allocated, calls will be ignored.  (We could
 * raise an error, but it seems better to make it a no-op, so that
 * libraries containing such calls can be reloaded if needed.)
 */
void
RequestAddinShmemSpace(Size size)
```